ubccr/xdmod
# Acl/config update

## Description
- Updated the way in which Config.php constructs an internal data representation
  of data. Specifically, as viewed in the context of which module added which pieces of data. This can be
  used with the 'filterByMetaData' function to retrieve a module centric view of
  the required config section ( honoring the 'extends' directives and other
  modules 'inheriting' from xdmod ).
- Updated acl-config to work with this new method of retrieving module specific
  data.

## Motivation and Context
- Simplify the way in which we retrieve module specific data from the Config.php class.

## Tests performed
Manual Tests were performed using the following docker containers:
- tas-tools-ext-01.ccr.xdmod.org/centos7-xdmod6.6.0:version6
  - Installed Module Groups:
     - Plain Open XDMoD:
        - OpenXDMoD
      - OpenXDMoD w/ additional modules: 
        - OpenXDMoD
        - App Kernels
        - SUPREMM

- tas-tools-ext-01.ccr.xdmod.org/centos7-xdmod-supremm6.6.0:version:version3
    - Installed Module Groups:
        - XSEDE XDMoD:
            - OpenXDMoD
            - XSEDE
            - App Kernels
            - SUPREMM

**Test 1**
- Navigate to XDMoD running on container
- Ensure that you see three tabs: Summary, Usage, About
- Click the `Usage` tab
- Ensure that the tab loads successfully
- Ensure that the realm groupings in the left side tree contains all of the realms that are installed that public should have access to.
- Log In as a user that has the acl `usr`
- Ensure that the following tabs are present
  - `Metric Explorer`
  - `Allocations`
  - `Report Generator`
- Depending on which modules are installed, the following tabs may also be present:
  - XSEDE:
    - `Compliance`
    - `Sci Impact`
  - App Kernels:
    - `App Kernels`
  - SUPREMM
    - `Job Viewer`
- Click a few of the tabs and make sure that things are working as expected

**Test 2**
- spawn a shell in the container being tested
- login to mysql
- Execute `use moddb;`
- Execute the following and ensure there are no duplicate records 
```sql
SELECT * FROM acls;
```
- Execute the following and ensure there are no duplicate records:
```sql
SELECT * FROM tabs;
```

- Execute the following and ensure there are records present:
```sql
SELECT * FROM acl_hierarchies;
```
- Execute  the following and sure that the counts of tabs / acl are as expected.
```sql
SELECT 
  a.name, 
  COUNT(a.acl_id) as num 
FROM acl_tabs at 
JOIN acls a ON a.acl_id = at.acl_id 
JOIN tabs t ON t.tab_id = at.tab_id 
GROUP BY a.acl_id;
```

## Types of changes


## Checklist:
